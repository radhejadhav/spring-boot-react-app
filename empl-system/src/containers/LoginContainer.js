import React, { useEffect } from 'react'
import LoginComponent from '../components/LoginComponent'
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { setErrors, tokenRequest } from '../actions/LoginAction'
import { loginSchema } from '../validation/ValidationSchema';
import { HOME } from '../shared/RouterConstant';

export default function LoginContainer() {

    const navigate = useNavigate();
    const dispatch = useDispatch()
    const data = useSelector(state => state.login)

    const { username, password, isSessionValid } = data;

    const onSubmitHandler = () => {
        loginSchema.isValid({ username, password }).then((valid) => {
            if (valid) {
                dispatch(tokenRequest({ username, password }))
            } else {
                loginSchema.validate({ username, password }, { abortEarly: false })
                    .catch(err => {
                        err.inner.forEach(element => {
                            dispatch(setErrors(element))
                        });
                    })
            }
        })
    }

    useEffect(()=>{
        if(isSessionValid){
            navigate(HOME)
        }
    },[isSessionValid])

    const onChangeHandler = (event) => {
        dispatch({ type: "usernameError", value: "" })
        dispatch({ type: "passwordError", value: "" })
        dispatch({ type: event.target.name, value: event.target.value })
        loginSchema.validate({ username, password }, { abortEarly: false })
            .catch(err => {
                err.inner.forEach(element => {
                    dispatch(setErrors(element))
                });
            })
    } 

    return (
        <LoginComponent
            onChangeHandler={onChangeHandler}
            onSubmitHandler={onSubmitHandler}
            dispatch={dispatch}
            state={data}
        />
    )
}
