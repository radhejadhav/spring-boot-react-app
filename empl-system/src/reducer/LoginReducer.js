import { IS_SESSION_VALID, PASSWORD, PASSWORD_ERROR, USERNAME, USERNAME_ERROR, USER_PROFILE } from "../shared/ReducerConstant"

export const initialState = {
    username: "",
    password: "",
    usernameError: "",
    passwordError: "",
    userProfile: {},
    isSessionValid: false
}

const LoginReducer = (state = initialState, action) => {
    switch (action.type) {
        case USERNAME:
            return { ...state, username: action.value }
        case PASSWORD:
            return { ...state, password: action.value }
        case USERNAME_ERROR:
            return { ...state, usernameError: action.value }
        case PASSWORD_ERROR:
            return { ...state, passwordError: action.value }
        case USER_PROFILE:
            return { ...state, userProfile: action.value }
        case IS_SESSION_VALID:
            return { ...state, isSessionValid: action.value }
        default:
            return state
    }

}

export default LoginReducer;
