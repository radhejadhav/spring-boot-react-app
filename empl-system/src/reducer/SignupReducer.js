import { COMPANY, CONFIRM_PASSWORD, DISPLAY_NAME, EMAIL_ID, FIELD_ERROR, FIRST_NAME, LAST_NAME, MOBILE, PASSWORD, SET_FIELD_ERROR, SIGNUP_RESPONSE, USER_NAME } from "../shared/ReducerConstant"

const initialState = {
    email: "",
    displayName: "",
    firstName: "",
    lastName: "",
    mobile: "",
    username: "",
    company: "",
    password: "",
    confirmPassword: "",
    fieldError: {},
    signupResponse: {}
}

const SignupReducer = (state = initialState, action) => {

    switch (action.type) {
        case EMAIL_ID:
            return { ...state, email: action.value }
        case FIRST_NAME:
            return { ...state, firstName: action.value }
        case LAST_NAME:
            return { ...state, lastName: action.value }
        case DISPLAY_NAME:
            return { ...state, displayName: action.value }
        case USER_NAME:
            return { ...state, username: action.value }
        case MOBILE:
            return { ...state, mobile: action.value }
        case COMPANY:
            return { ...state, company: action.value }
        case PASSWORD:
            return { ...state, password: action.value }
        case CONFIRM_PASSWORD:
            return { ...state, confirmPassword: action.value }
        case FIELD_ERROR:
            return { ...state, fieldError: { ...state.fieldError, [action.value.name]: action.value.value } }
        case SET_FIELD_ERROR:
            return { ...state, fieldError: { ...undefined } }
        case SIGNUP_RESPONSE:
            return {...state, signupResponse:{...action.value}}
        default:
            return state;
    }
}

export default SignupReducer;