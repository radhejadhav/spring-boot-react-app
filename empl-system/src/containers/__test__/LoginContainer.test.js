import { fireEvent, render, screen } from "@testing-library/react"
import { Provider } from "react-redux"
import { BrowserRouter } from "react-router-dom"
import { applyMiddleware, combineReducers, createStore } from "redux"
import LoginReducer from "../../reducer/LoginReducer"
import SignupReducer from "../../reducer/SignupReducer"
import LoginContainer from "../LoginContainer"

describe("Test Login Container", () => {
    let fragment, getText

    beforeEach(() => {
        const { asFragment, getByText } = render(
            <Provider store={createStore(combineReducers({ login: LoginReducer, signup: SignupReducer }))}>
                <BrowserRouter>
                    <LoginContainer />
                </BrowserRouter>
            </Provider>
        )
        fragment = asFragment
        getText=getByText
    })

    it("Should return LoginComponent", () => {
        expect(fragment()).toMatchSnapshot()
    })
    // it("Should validate the first name field", () => {
    //     fireEvent.click(getText("button"))
    // })
})