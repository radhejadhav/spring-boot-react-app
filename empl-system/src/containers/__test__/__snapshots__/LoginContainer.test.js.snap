// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Container Should return LoginComponent 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="my-4 col-sm-12 col-md-6 offset-md-3"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            Login
          </div>
          <div
            class="card-body"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
              >
                Username
              </label>
              <input
                aria-invalid="false"
                class="form-control"
                name="username"
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
              >
                Password
              </label>
              <input
                aria-invalid="false"
                class="form-control"
                name="password"
                type="password"
                value=""
              />
              <div
                class="invalid-feedback"
              />
            </div>
            <div
              class="mb-3"
            >
              Not an account ? 
              <a
                href="/register"
              >
                Create Here..!
              </a>
              <br />
            </div>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
